hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -2102,15 +2102,22 @@ public void run() {
         }
 
         private void doRun() throws InterruptedException {
-            for (; ; ) {
+            for (; ;) {
                 if (!isMigrationAllowed()) {
                     break;
                 }
-                Runnable r = migrationQueue.poll(1, TimeUnit.SECONDS);
+
+                Runnable r = migrationQueue.peek();
                 if (r == null) {
+                    Thread.sleep(100);
```

Comment:
We are not checking 10 more frequent than before? Because initially the frequency was 1/second and now 10/second.